onnx/reference/ops/op_reduce_sum.py

```diff
@@ -15,20 +13,10 @@ def _run(self, x, axes=None, keepdims=None):  # type: ignore # pylint: disable=W
 
 
 class ReduceSum_13(OpRunReduceNumpy):
-    def run(self, x, axes=None, keepdims=None):  # type: ignore
-        keepdims = keepdims or self.keepdims  # type: ignore
-        res = self._run(x, axes=axes, keepdims=keepdims)
-        if res[0].dtype != x.dtype:
-            raise RuntimeTypeError(
-                f"Output type mismatch: input {x.dtype} != output {res[0].dtype} "
-                f"(operator {self.__class__.__name__!r})."
-            )
-        return res
-
-    def _run(self, x, axes=None, keepdims=None):  # type: ignore
+    def _run(self, x, axes=None, keepdims=None, noop_with_empty_axes=None):  # type: ignore
         if (
             axes is None or len(axes.shape) == 0 or axes.shape[0] == 0
-        ) and self.noop_with_empty_axes:  # type: ignore
+        ) and noop_with_empty_axes:  # type: ignore
```

Comment:
Wondering why there is a condition `len(axes.shape) == 0` in previous line? This may have nothing to do with the issue this PR is fixing, but just trying to understand. I believe len(axes.shape) == 0 means a zero-dimensional tensor (which is a scalar with one value?

Reply:
One unit test in the backend test defines the axes as `axes = np.array([], dtype=np.int64)`. shape is `(0,)`. I'll improve the code.